### DISCUSSION TITLE
WithSpan usage in the context of asynchronous GRPC calls

### DISCUSSION THREAD

**Discussion by itsmaneesh** [open-telemetry/opentelemetry-java-instrumentation#5708] [Q&A]
Hi team,
I have a requirement to use async (future) stub type to make non-blocking calls to GRPC server. In order to have a custom span in this non-blocking calls case, I can use @WIthSpan annotation. But, I wanted to understand whether @WithSpan is able to handle the async calls in maintaining the Context.? If so, how it is being handle (any pointers to the documentation / explanation will be great).

Thank you.

**Comment by trask** [comment]
hi @itsmaneesh, just want to first check why the spans produced by grpc auto-instrumentation are not sufficient for your use case?

**Reply by itsmaneesh** [reply]
Hi @trask , We have a requirement where a method A (that performs a certain operations) need to be invoked from GRpcService method.
Like you mentioned, yes auto-instrumentation produces a span for the GRpcService method, However, it does NOT create a child span for the method A.
So, I wanted to use @WithSpan annotation at this method A level to have my own custom span as a child span. 
Please let me know does this clarify you question.

Can you please provide answer to my initial post's question.

Thank you.

**Reply by trask** [reply]
check out https://github.com/open-telemetry/opentelemetry.io/pull/1248/files#diff-295191b54e0f5b8219639eecc82a787ead91709a78d5cb7c64ce80c65293cc3b. except for those cases, the span will be ended when the method returns
